### DISCUSSION TITLE
VideoCapture using CamIndex

### DISCUSSION THREAD

**Discussion by AnalogBrain** [emgucv/emgucv#419] [Q&A]
Hello all,

I'm utilizing Emgu CV in a pretty simple way to get a live video feed from one of two different cameras, which I switch between in code.  However, I recently noticed that the camIndex values of my cameras changed on me, possibly from an update?  Before, index 0 was the laptop webcam, and indexes 1 and 2 were my cameras, but it just recently switched from indexes 0 and 1 being my cameras, with index 3 being the laptop webcam.  Is there a better way to grab the cameras I want other than using camIndex, since it seems to have changed on me out of the blue?  Thanks!

**Comment by emgucv** [comment]
Emgu CV's VideoCapture class calls Open CV's VideoCapture class, which in term, use the various system capture backend to create camera capture objects. 

One windows, the default VideoCapture backend is MSMF (MicroSoft Media Foundation). It is MSMF that assign camera index to camera and it will need to be fixed on the MSMF side. I don't think Open CV has option to open camera other than using camIndex. You may create a ticket in Open CV to request creating camera capture other than using index. If it is added in Open CV we will also include that feature in Emgu CV.

Optionally, on windows, you can try to create a VideoCapture device using DSHOW (direct show) backend. It may have a different implementation when indexing camera, compares to MSMF.

**Reply by AnalogBrain** [reply]
So say if I used the DirectShow Library for C# to get a list of cameras and then used the DSHOW backend for VideoCapture, the indexes for each should match!  Seems to work now using that, thanks for your help!

**Reply by emgucv** [reply]
Very cool work around! Thanks for letting us know.
